Rewrite xiaomi-ble documentation (#31227)
home-assistant/home-assistant.github.io: source/_integrations/xiaomi_ble.markdown

ha_integration_type: integration
---

> Integrates devices that implement the Xiaomi Mijia BLE MiBeacon protocol and the BLE protocols implemented in the MiScales and the MiFora plant sensor. It listens to Bluetooth broadcasts that the device makes by itself, allowing us to track the latest sensor values or events without needing to wake it up from deep sleep to poll and conserving its battery power.

Reviewer: ```suggestion
Integrates devices that implement the Xiaomi Mijia BLE MiBeacon protocol and the BLE protocols implemented in the MiScales and the MiFora plant sensor. The integration listens to Bluetooth broadcasts that the device makes by itself, allowing it to track the latest sensor values or events without waking the device up for polling.  This method conserves battery lifetime.
```

> The integration will automatically discover devices once the [Bluetooth](/integrations/bluetooth) integration is enabled and functional. The entities are added after the values are first received. This means entities for values that are broadcasted at a lower interval (e.g., battery) might show up later.

Reviewer: ```suggestion
The integration automatically discovers devices once the [Bluetooth](/integrations/bluetooth) integration is enabled and functional. The entities are added after the values are first received. This means that entities might show up later if the corresponding values are broadcasted at a lower interval (for example, battery).
```

There are a few ways to obtain a bindkey for your device:

- Extract the keys from Xiaomi Cloud using a [token extractor](https://github.com/PiotrMachowski/Xiaomi-cloud-tokens-extractor) tool.

> - Set your own. The [Telink Flasher](https://pvvx.github.io/ATC_MiThermometer/TelinkMiFlasher.html) by pvvx allows you to generate new bindkeys for devices it supports (LYWSD03MMC, MJWSD05MMC, MHO-C401, CGG1, and CGDK2). This online flashing tool also allows you to upload alternative firmware, with some improvements compared to the stock firmware, like faster sensor updates. Note that the new bind key will work with Home Assistant, but the Mi Home app will not recognize the sensor anymore once the device has been activated by the TeLink flasher application. To use the sensor again with the Xiaomi Mi Home app, the device needs to be removed and then re-added inside the Mi Home app.

Reviewer: ```suggestion
- Set your own. The [Telink Flasher](https://pvvx.github.io/ATC_MiThermometer/TelinkMiFlasher.html) by pvvx allows you to generate new bindkeys for devices it supports (LYWSD03MMC, MJWSD05MMC, MHO-C401, CGG1, and CGDK2). This online flashing tool also allows you to upload alternative firmware, with some improvements compared to the stock firmware, like faster sensor updates. Note that the new bind key works with Home Assistant, but the Mi Home app does not recognize the sensor anymore once the device has been activated by the TeLink flasher application. To use the sensor again with the Xiaomi Mi Home app, the device needs to be removed and then re-added inside the Mi Home app.
```

- An alternative [Telink Flasher](https://atc1441.github.io/Temp_universal_mi_activate.html) by atc1441 also allows you to generate new bind keys, and supports even more Xiaomi devices compared to the Telink Flasher by pvvx.

> - Yeelight Remote (YLYK01YL) and dimmers (YLKG07YL and YLKG08YL) can use V2/V3 MiBeacon encryption, although more recent devices seem to be supplied with V4/V5 encryption. For V2/V3 MiBeacon encryption, the bindkey can't be be determined with the above methods. Instructions on how to get the bindkey for these devices can be found in the [BLE monitor FAQ](https://custom-components.github.io/ble_monitor/faq#how-to-get-the-mibeacon-v2v3-encryption-key).

Reviewer: ```suggestion
- Yeelight Remote (YLYK01YL) and dimmers (YLKG07YL and YLKG08YL) can use V2/V3 MiBeacon encryption, although more recent devices seem to be supplied with V4/V5 encryption. For V2/V3 MiBeacon encryption, the bindkey can't be determined with the above methods. Instructions on how to get the bindkey for these devices can be found in the [BLE monitor FAQ](https://custom-components.github.io/ble_monitor/faq#how-to-get-the-mibeacon-v2v3-encryption-key).
```
